### PR TITLE
Fix permission screen & trailing zeros

### PR DESCRIPTION
**Pull request by cammellos**
There were a few issues with the permission screen:
1) Wrong permission was displayed when able to join
2) If not able to join, we were showing both admin/member permissions
3) Trailing zeros in token amount

### CODE REVIEW

**Comment by pavloburykh** [comment]
@cammellos thanx for the PR. Please take a look at the issue

### ISSUE 1 User is eligible to join token gated community despite does not meet requirements

**Steps:** 

1. User A (Desktop Admіn) creates token gated community
2. User A shares community to User B who does not meet membership requirements
3. See if User B is eligible to join the community

**Expected result:** User B is not eligible to join

**Actual result:** User B is eligible to join and send community request. After receiving the request Desktop admin will see an alert (see screeshot below). 

[Status-debug-logs - 2024-02-08T183109.774.zip](https://github.com/status-im/status-mobile/files/14211998/Status-debug-logs.-.2024-02-08T183109.774.zip)

https://github.com/status-im/status-mobile/assets/97245802/54065304-0a4c-4a46-9e5a-752684218a7e

Status Desktop 2024-02-08 18-30-07

File: src/status_im/config.cljs [hunk 1]
```diff
@@ -167,5 +167,5 @@
 
 (def default-kdf-iterations 3200)
 
-(def community-accounts-selection-enabled? (enabled? (get-config :ACCOUNT_SELECTION_ENABLED "0")))
```
Comment (cammellos) [inline src/status_im/config.cljs:170]: will remove this before merging

**Comment by pavloburykh** [comment]
@cammellos thanx for the PR. ISSUE 1 is fixed. Please take a look at another issues. 

### ISSUE 2 Permissions are not shown for token gated community when user is eligible to join as member

**Steps:** 

1. User A shares token gated community to User B (i.e. 1 ETH to become a member)
2. User B satisfies the requirements as he holds 1 ETH
3. See if token requirements are displayed for User B on community home screen

**Expected result:** permissions should be shown with text "You are eligible to join, you hodl:..."

[Figma](https://www.figma.com/file/h9wo4GipgZURbqqr1vShFN/Communities-for-Mobile?type=design&node-id=15331-284865&mode=design&t=ddL4aBx6CDlSLcoh-0) 

Communities for Mobile – Figma 2024-02-12 13-53-47

**Actual result:** permissions are not shown. 

[Status-debug-logs - 2024-02-12T140714.393.zip](https://github.com/status-im/status-mobile/files/14238154/Status-debug-logs.-.2024-02-12T140714.393.zip)

Screen of permissions from Desktop:

Status Desktop 2024-02-12 13-21-19

Screen of community home screen from mobile:
photo_2024-02-12 14 05 38
photo_2024-02-12 14 05 41

**Comment by pavloburykh** [comment]
### ISSUE 3 Mobile user can send request to join token gated community despite he does not meet requirements anymore

**Steps:** 

1. User A shares token gated community to User B (i.e. 1 ETH to become a member)
2. User B satisfies the requirements as he holds 1 ETH
3. User B does not send community request
4. User A changes requirements so that User B is no longer eligible to join the community
5. See if requirements have been updated on User's B side

**Expected result:** requirements are updated, User B cannot send community request as he is not eligible to join. UI should be updated to the according state 

**Actual result:** User B still able to send community request, as Send request button is active. No info about User B is not eligible to join. 

You can user same logs from issue 1 for debugging. 

Screenshot of permissions from desktop: 
Status Desktop 2024-02-12 14-16-44

Screenshot from mobile User B:

photo_2024-02-12 14 18 19
photo_2024-02-12 14 18 22

File: src/status_im/contexts/communities/overview/view.cljs [hunk 2]
```diff
@@ -110,66 +110,70 @@
...
-     (when (pos? number-of-hold-tokens)
+    (if (or (not role-permissions?) no-member-permission?)
+      [request-access-button id color]
+      [rn/view {:style (style/token-gated-container)}
+       [rn/view
+        {:style {:padding-horizontal 12
+                 :flex-direction     :row
+                 :align-items        :center
+                 :justify-content    :space-between
+                 :flex               1}}
```
Comment (ajayesivan) [inline src/status_im/contexts/communities/overview/view.cljs:154]: Could you move this style to the style file

**Review by ajayesivan — COMMENTED** [review]
From what I remember from previous discussions, we should only show admin & token master requirements if the user meets them otherwise it should be just member permissions. Correct me if I'm wrong.

Screenshot 2024-02-12 at 18 46 37



Also, 'You must hodl:' text is missing when the user doesn't meet any requirements. See design: https://www.figma.com/file/h9wo4GipgZURbqqr1vShFN/Communities-for-Mobile?type=design&node-id=11684-427664&mode=design&t=3LdUNA15MNynYyvj-4

**Comment by pavloburykh** [comment]
> From what I remember from previous discussions, we should only show admin & token master requirements if the user meets them otherwise it should be just member permissions. Correct me if I'm wrong. 

@ajayesivan are those screens captured in builds from this PR? The issues you are pointing out should be fixed in recent builds.  

Also, I believe that for creating community and managing permissions you should use Desktop build that includes corresponding go changes https://github.com/status-im/status-desktop/pull/13240

**Comment by ajayesivan** [comment]
I was testing locally and missed that this PR has status-go changes. After rebuilding, everything is working fine. Thank you @pavloburykh.

**Review by ajayesivan — APPROVED** [review]

**Comment by yevh-berdnyk** [comment]
hi @cammellos 
I've just pushed a commit with e2e updates, restarting the tests

**Pull request merged**